src/main/java/chess/domain/Board.java

```diff
@@ -18,6 +20,56 @@ public Board() {
         initializePiece();
     }
 
+    public Map<Position, String> move(Position currentPosition, Position nextPosition, Color thisTurn) {
+        Piece piece = board.getOrDefault(currentPosition, BlankPiece.getInstance());
+        List<Position> routePositions = piece.move(currentPosition, nextPosition);
+        validateThisTurnColor(thisTurn, piece);
+
+        if (piece.isPawn()) {
+            return movePawn(currentPosition, nextPosition, routePositions);
+        }
+
+        return moveGeneralPiece(nextPosition, piece, routePositions);
+    }
+
+    private void validateThisTurnColor(Color thisTurn, Piece piece) {
+        if (piece.isSameColor(thisTurn) == false) {
+            throw new IllegalArgumentException("이번 차례에 움직일 수 있는 색의 기물이 아닙니다.");
+        }
```

Comment:
코드를 아래처럼 바꿔봐도 좋을것 같아요 😉
```suggestion
       if (!piece.isSameColor(thisTurn)) {
            throw new IllegalArgumentException("이번 차례에 움직일 수 있는 색의 기물이 아닙니다.");
        }
```

Reply:
이 부분은 원래는 저도 아래의 예시처럼 ! 을 사용했었는데,
개인적으로 이 !가 눈에 띄지 않아 읽기 헷갈린다는 느낌을 받아 == false를 사용했습니다.
그럼에도 대부분의 타 개발자 독자는 !로 읽는 게 편할테니 수정하는 게 맞다는 의도이신가요?
일단 !를 사용하도록 수정하겠습니다